packages/vue-app/template/utils.js

```diff
@@ -146,7 +146,19 @@ export function resolveRouteComponents (route, fn) {
     flatMapComponents(route, async (Component, instance, match, key) => {
       // If component is a function, resolve it
       if (typeof Component === 'function' && !Component.options) {
-        Component = await Component()
+        try {
+          Component = await Component()
+        } catch (error) {
+          // Handle webpack chunk loading errors
+          // This may be due to a new deployment or a network problem
+          if (error && error.name === 'ChunkLoadError' && window.location.hash !== '#retry') {
+            // mark the page not to reload infinitely
+            window.location.hash = '#retry'
```

Comment:
Modification of URL, might interference with application logic. We may use [navigation type](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation/type) guard. (safari workaround needed)